### ISSUE TITLE
HTTP header matching should be case insensitive to support more clients

### ISSUE THREAD

**Issue by nibarius** [davidepianca98/KMQTT#13]
Some MQTT clients, such as [HiveMQ](https://github.com/hivemq/hivemq-mqtt-client) uses all lowercase in the HTTP headers during a websocket handshake.

KMQTT does a case sensitive regex match against, for example, [`Sec-WebSocket-Protocol`](https://github.com/davidepianca98/KMQTT/blob/master/src/commonMain/kotlin/socket/tcp/WebSocket.kt#L46). Due to this it's not possible to use WebSockets to connect to KMQTT from a client using on HiveMQ.

It would be great if case insensitive matching could be used when matching against http headers.

For reference, here's an example of what the request and response looks like during the handshake:

```
GET / HTTP/1.1
host: 192.168.22.52:1884
upgrade: websocket
connection: upgrade
sec-websocket-key: GDDymPLXlAntP0fSTArBCg==
origin: http://192.168.22.52:1884
sec-websocket-protocol: mqtt
sec-websocket-version: 13

HTTP/1.1 400 Bad Request
```

**Comment by davidepianca98** [comment]
Hello, thank you very much for letting me know, I just released version 0.3.2 that should fix this issue. If you still find anything wrong please feel free to reopen.

**Comment by nibarius** [comment]
I tested 0.3.2 and it's working fine together with HiveMQ. Thanks.
